bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoHandler.java

```diff
@@ -391,44 +411,48 @@ public String getUDN() {
      *
      */
     protected void updateWemoState() {
-        String action = "GetBinaryState";
-        String variable = "BinaryState";
         String actionService = "basicevent";
-        String value = null;
-
-        if (getThing().getThingTypeUID().getId().equals("insight")) {
-            action = "GetInsightParams";
-            variable = "InsightParams";
-            actionService = "insight";
-        }
-
-        String soapHeader = "\"urn:Belkin:service:" + actionService + ":1#" + action + "\"";
-        String content = "<?xml version=\"1.0\"?>"
-                + "<s:Envelope xmlns:s=\"http://schemas.xmlsoap.org/soap/envelope/\" s:encodingStyle=\"http://schemas.xmlsoap.org/soap/encoding/\">"
-                + "<s:Body>" + "<u:" + action + " xmlns:u=\"urn:Belkin:service:" + actionService + ":1\">" + "</u:"
-                + action + ">" + "</s:Body>" + "</s:Envelope>";
+        if (host != null && !host.isEmpty()) {
+            String wemoURL = getWemoURL(host, actionService);
+            if (wemoURL != null) {
```

Comment:
No error is logged if wemoURL is null here. I would propose to exit fast to reduce nesting and increase readability as in a previous comment. This would make such problem more noticeable as it's somewhat hidden in all this nesting/condition checking.

Reply:
refactored